server/src/main/java/org/spine3/server/stand/AggregateStateIdStringifier.java

```diff
@@ -0,0 +1,181 @@
+ *
+ * <p>The protocol of strigification is as follows:
+ * {@code {@linkplain TypeUrl}_of_the_state_type}-{@code ID_TYPE_NAME_or_SHORT_REPR_FOR_PRIMITIVES}-{@code generic_ID_stringifier_output}
+ *
+ * @author Dmytro Dashenkov
+ */
+class AggregateStateIdStringifier extends Stringifier<AggregateStateId> {
+
+    private static final String DIVIDER = "-";
+    private static final int MEAN_STRING_LENGTH = 256;
+
+    @Override
+    protected String toString(AggregateStateId id) {
+        checkNotNull(id);
+
+        final String typeUrl = id.getStateType()
+                                 .value();
+        final Object genericId = id.getAggregateId();
+        final Class genericIdType = genericId.getClass();
+        final String idTypeString = idTypeToString(genericIdType);
+        final String genericIdString = Stringifiers.toString(genericId);
+
+        final String result = new StringBuilder(MEAN_STRING_LENGTH).append(typeUrl)
+                                                                   .append(DIVIDER)
+                                                                   .append(idTypeString)
+                                                                   .append(DIVIDER)
+                                                                   .append(genericIdString)
+                                                                   .toString();
+        return result;
+    }
+
+    @Override
+    protected AggregateStateId fromString(String s) {
+        checkNotNull(s);
+
+        final int typeUrlEndIndex = s.indexOf(DIVIDER);
+        checkArgument(typeUrlEndIndex > 0,
+                      "Passed string does not contain a type URL.");
+        final String typeUrlString = s.substring(0, typeUrlEndIndex);
+        final TypeUrl typeUrl = TypeUrl.parse(typeUrlString);
+
+        final int idTypeStartIndex = typeUrlEndIndex + 1;
+        final int idTypeEndIndex = s.indexOf(DIVIDER, idTypeStartIndex);
+        checkArgument(typeUrlEndIndex > 0,
+                      "Passed string does not contain the ID type.");
+        final String idTypeString = s.substring(idTypeStartIndex, idTypeEndIndex);
+        final Class idType = idTypeFromString(idTypeString);
+
+        final String idStringValue = s.substring(idTypeEndIndex + 1);
+        final Object genericId = Stringifiers.fromString(idStringValue, idType);
+
+        final AggregateStateId result = AggregateStateId.of(genericId, typeUrl);
+
+        return result;
+    }
+
+    private static String idTypeToString(Class idType) {
+        checkSupported(idType);
+        final IdType type = IdType.get(idType);
+        final String typeString = type.describe(idType);
+        return typeString;
+    }
+
+    private static Class idTypeFromString(String idTypeString) {
+        final Class type = IdType.getTypeFrom(idTypeString);
+        return type;
+    }
+
+    /**
+     * An {@code enum} of allowed ID types.
+     */
+    private enum IdType {
```

Comment:
As discussed vocally, let's avoid duplicating the enum of supported ID types (see `Identifier.Type`).